### PR TITLE
[documentation] #1685, #1694: Update [Contributing.md].

### LINKED ISSUE #1694: Update crates versions in rc1 branch
Crates versions should be `2.0.0-pre.1` instead of current `0.1.0`. As this is the version that will be released to crates.io

### PR DESCRIPTION
**Pull request by appetrosyan**
### Description of the Change

Updated `contributing.md` in accordance with points discussed at the `Iroha2::Core` meeting on 2021-12-08 Wed. 

### Issue

Partially addresses #1685 
Closes #1694 

### Benefits
More up-to-date and enforceable contributing guide. 

### Possible Drawbacks

None

### CODE REVIEW

File: .github/issue_template.md [hunk 1]
```diff
@@ -1,7 +1,36 @@
-## Hello! 
+<!-- IMPORTANT: add the `iroha2` for IROHA2-related issues -->
```
Comment (mversic) [inline .github/issue_template.md:1]: note to add `iroha2` label?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,144 +8,191 @@ The following is a short set of guidelines for contributing to Iroha.
...
-- impl `trait-from-std` for struct
-- submodules with bodies
-- pub mod prelude
-
-### Documentation Styleguide
-
--  Use `Rust Docs`
-
-## Places where community is active
+#### Keep (with tweaks)
```
Comment (mversic) [inline CONTRIBUTING.md:155]: is this a leftover?

File: .github/issue_template.md (hunk 1 above)
Comment (appetrosyan) [reply .github/issue_template.md:1]: Yep. I personally forget to add the label.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -8,144 +8,189 @@ The following is a short set of guidelines for contributing to Iroha.
...
+- Group related functionality spatially, preferably inside appropriate modules. For example, instead of having a block with `struct` definitions and then `impl`s for each individual struct, it is better to have the `impl`s related to that `struct` next to it.
+- Otherwise:  declaration before implementation;`use` statements at the top.
+- When writing non-doc comments, instead of describing *what* your function does, try to explain *why* it does something in a particular way. This will save you and the reviewer time.
+- Try to avoid `use` statements if the imported name is used only once. This makes moving your code into a different file easier.
+- Do not silence clippy lints indiscriminately. If you do, explain your reasoning with a comment (or `expect` message).
+- We use pinned dependencies. If your work depends on a particular crate, see if it wasn't already installed using [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html) (hint use `bat` or `grep`), and try to use that version, instead of the latest version.
+- Version bumps in separate PR.
```
Comment (mversic) [inline CONTRIBUTING.md:177]: could you add that we prefer full format(`"x.y.z"`) for our dependency versions? as opposed to `"x.y"` or `"x"`. This is actually implicitly caret requirement like `"^x.y.z"` and will make `cargo update` bump both patch and minor versions

I think this alone is sufficient and we should only allow other version requirements in justified special cases

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -8,144 +8,189 @@ The following is a short set of guidelines for contributing to Iroha.
...
+  - The name is an accepted abbreviation (e.g. `tx`, `wsv` etc) TODO link glossary.
+  - The full name would have been shadowed by a local variable (e.g. `msg <- message`).
+  - The full name would have made the code cumbersome (no more than 5-6 words in name) (e.g. `WorldStateViewReceiverTrait -> WSVRecvTrait`).
+  - If in doubt: clearer is better than shorter.
+- Never ignore an error. If you can't panic and can't recover, it at least needs to be recorded in the log.
+- Prefer to return a `Result` instead of `panic!`.
+  - Except when implementing something that uses `issue_send` instead of `send` ([more about actors](docs/source/guides/actor.md)). Actors and parallelism don't mix; you could deadlock the entire peer, so it's better to `panic!` if something goes wrong. This is a necessary concession for asynchronous programming.
+- Group related functionality spatially, preferably inside appropriate modules. For example, instead of having a block with `struct` definitions and then `impl`s for each individual struct, it is better to have the `impl`s related to that `struct` next to it.
+- Otherwise:  declaration before implementation;`use` statements at the top.
```
Comment (mversic) [inline CONTRIBUTING.md:172]: module imports and constants at the top?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -8,144 +8,189 @@ The following is a short set of guidelines for contributing to Iroha.
...
 
 ### Rust Style Guide
 
-- Use `cargo fmt --all`
-- Prefer using `mod.rs` inside the module directory to the `.rs` file named the same as the module in the top level directory.
-- Use domain-first modules structure. For example `domain::isi::*`. Such a way
-complex uses will be easier to incude in dependent modules.
-- Do not use whitespaces or empty lines inside function bodies.
-- Put public methods first in your impl blocks.
-- Put inner modules after `self` module content, but before `tests` module.
```
Comment (mversic) [inline CONTRIBUTING.md:122]: we don't want this anymore?

Comment (appetrosyan) [reply CONTRIBUTING.md:122]: Easy. The re-ordering of modules is handled by `cargo fmt`, `tests` at the end is standard practice in `rust` code. Whitespace in `fn` bodies is not always bad.

File: .github/issue_template.md [hunk 6]
```diff
@@ -1,7 +1,36 @@
...
+<!-- Explain what's wrong in your own words. -->
 
-Still, if it is more convenient for you to create an issue on GitHub, our Yozhik Bot will carefully transfer it to JIRA and close the issue here. 
+
+
+# Minimum working example
+
+## Given
+
+<!-- What I started with -->
```
Comment (eadventurous) [inline .github/issue_template.md:15]: If it is a bug, we should ask for environment details like iroha version or commit, how many peers, config, etc.

File: CONTRIBUTING.md (hunk 5 above)
Comment (mversic) [reply CONTRIBUTING.md:122]: I was just referring to the last item, but ok

File: .github/issue_template.md (hunk 6 above)
Comment (appetrosyan) [reply .github/issue_template.md:15]: Done.

**Review by mversic — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**